packages/app/src/cli/services/generate.ts

```diff
@@ -90,31 +79,36 @@ async function buildPromptOptions(
   }
 }
 
-function checkLimits(templateSpecifications: TemplateSpecification[], app: AppInterface) {
+function checkLimits(
+  templateSpecifications: TemplateSpecification[],
+  specifications: GenericSpecification[],
+  app: AppInterface,
+) {
   const iterateeFunction = (spec: TemplateSpecification) => {
-    const allValid = spec.types.every((type) => !limitReached(app, type))
+    const allValid = spec.types.every((type) => !limitReached(app, specifications, type))
     return allValid ? 'validTemplateSpecifications' : 'templatesOverlimit'
   }
   return groupBy(templateSpecifications, iterateeFunction)
 }
 
-function limitReached(app: AppInterface, type: GenericSpecification) {
-  if (isFunctionExtension(type)) {
+function limitReached(app: AppInterface, specifications: GenericSpecification[], templateType: TemplateType) {
+  const type = templateType.type
+  if (type === 'function') {
     return app.extensions.function.length >= blocks.functions.defaultRegistrationLimit
   } else {
-    return app.extensionsForType(type).length >= type.registrationLimit
+    const specification = specifications.find((spec) => spec.identifier === type || spec.externalIdentifier === type)
+    const existingExtensions = app.extensionsForType({identifier: type, externalIdentifier: type})
+    return existingExtensions.length >= (specification?.registrationLimit || 1)
   }
 }
 
 async function saveAnalyticsMetadata(promptAnswers: GenerateExtensionPromptOutput, typeFlag: string | undefined) {
   await Promise.all(
     promptAnswers.extensionContent.map((extensionContent) => {
       return metadata.addPublicMetadata(() => ({
-        cmd_scaffold_template_flavor: extensionContent.extensionFlavor,
-        cmd_scaffold_type: extensionContent.specification.identifier,
-        cmd_scaffold_type_category: extensionContent.specification.category(),
```

Comment:
do we want to track this somehow? categories don't apply anymore, but maybe we want to track something here?